File: js/swap-nodes-paris/swap-nodes-paris.js
function ListNode(val, next) {
    this.val = (val===undefined ? 0 : val)
    this.next = (next===undefined ? null : next)
}

function swapPairs(head){
    let dummy = new ListNode(0, head)
    let prev = dummy, cur = head

    while (cur && cur.next) {
        let npn = cur.next.next
        let second = cur.next

        second.next = cur
        cur.next = npn
        prev.next = second

        prev = cur
        cur = npn
    }

    return dummy.next
}

let head = new ListNode(1)
let nodeA = new ListNode(2)
let nodeB = new ListNode(3)
let nodeC = new ListNode(4)

head.next = nodeA
nodeA.next = nodeB
nodeB.next = nodeC

res = swapPairs(head)

console.log(res)

File: go/swap-nodes-paris/swap-nodes-paris.go
package main

import(
	"fmt"
)

type ListNode struct {
    Val int
    Next *ListNode
}

func swapPairs(head *ListNode) *ListNode{
	if head == nil || head.Next == nil {
		return head
	}

	head, head.Next, head.Next.Next = head.Next, swapPairs(head.Next.Next), head

	return head
}

func main(){
	head := &ListNode{Val: 1}
	nodeA := &ListNode{Val: 2}
	nodeB := &ListNode{Val: 3}
	nodeC := &ListNode{Val: 4}

	head.Next = nodeA
	nodeA.Next = nodeB
	nodeB.Next = nodeC

	res := swapPairs(head)

	fmt.Printf("Swaping the nodes in %v the result is %v", head, res)

}